### PR TITLE
Feature/61186 move generic into commons

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/61186

This PR:

* "moves" classes creating extensions in common
* deprecates some filters and actions no more TEC related.

### CODE REVIEW

**Comment by tr1b0t** [comment]
:bangbang: This Pull Request is against a Central ticket for "Next Release". It shouldn't be merged until the Central ticket has been changed to "Current Release" or "CR Maintenance Release".

File: src/Tribe/Rewrite.php [hunk 1]
```diff
@@ -2,24 +2,18 @@
...
-		 * Static singleton variable
+	 * Static singleton variable
 	 * @var self
-		 */
-		public static $instance;
+	 */
+	public static $instance;
 
-		/**
+	/**
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: This line is tabbed correctly but much of what follows is still indented by one extra tab.

File: src/Tribe/Rewrite.php [hunk 2]
```diff
@@ -53,51 +47,25 @@ public function __construct(WP_Rewrite $wp_rewrite = null) {
 		}
 
 		/**
-		 * Static Singleton Factory Method
-		 *
+	 * Static Singleton Factory Method
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: Funky initial indentation in this docblock also.

File: src/Tribe/Rewrite.php [hunk 3]
```diff
@@ -53,51 +47,25 @@ public function __construct(WP_Rewrite $wp_rewrite = null) {
...
-		public function filter_generate( WP_Rewrite $wp_rewrite ) {
-			// Gets the rewrite bases and completes any other required setup work
-			$this->setup( $wp_rewrite );
+	/**
+	 * Generate the Rewrite Rules
+	 *
+	 * @param  WP_Rewrite $wp_rewrite WordPress Rewrite that will be modified, pass it by reference (&$wp_rewrite)
+	 */
+	public function filter_generate( WP_Rewrite $wp_rewrite ) {
+		parent::filter_generate($wp_rewrite);
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: Whitespace! (inside the parentheses)

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Rewrite.php:45](https://github.com/moderntribe/the-events-calendar/blob/094483ddf8c64154227dfac42e9be8fde2079728/src/Tribe/Rewrite.php#L45)
:x: | There must be a single space after the first parenthesis | [Rewrite.php:45](https://github.com/moderntribe/the-events-calendar/blob/094483ddf8c64154227dfac42e9be8fde2079728/src/Tribe/Rewrite.php#L45)
:x: | There must be a single space after the first parenthesis | [Rewrite.php:45](https://github.com/moderntribe/the-events-calendar/blob/094483ddf8c64154227dfac42e9be8fde2079728/src/Tribe/Rewrite.php#L45)
:x: | There must be a single space before the last parenthesis | [Rewrite.php:45](https://github.com/moderntribe/the-events-calendar/blob/094483ddf8c64154227dfac42e9be8fde2079728/src/Tribe/Rewrite.php#L45)

> via jenkins/codesniffer-phpcs

**Comment by barryhughes** [comment]
:v:

**Pull request merged**